### PR TITLE
Add tests for users and books

### PR DESCRIPTION
**Pull request by lucasenriquez**
## Summary

- [ ] I have added tests for users and books
- [ ] CircleCI run this tests
- [ ] BEFORE MERGE: I have updated the TRELLO card column before merging the pull request

## TRELLO Card

https://trello.com/c/8Q0KEkVZ/23-crear-tests-para-el-modelo-de-book

### CODE REVIEW

File: spec/factories/books.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :book do
+
```
Comment (melissaregnier) [inline spec/factories/books.rb]: If you create a factory, then it should create a valid Book instance. Try creating one using Faker gem.

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :user do
+
```
Comment (melissaregnier) [inline spec/factories/users.rb]: Same here.

File: spec/models/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+RSpec.describe Book, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: I don't think this should be here.

File: spec/models/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+require 'rails_helper'
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: You are requiring this twice.

File: spec/models/book_spec.rb [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+require 'faker'
+
+describe Book, type: :model do
+  it do
+    should validate_presence_of(:title)
+  end
+  it do
```
Comment (melissaregnier) [inline spec/models/book_spec.rb:8]: Always leave a new line between 'it' blocks

File: spec/models/book_spec.rb [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  end
+
+  subject(:book) do
+    Book.new(
+      genre: genre, author: author, image: image, title: title,
+      publisher: publisher, year: year
+    )
+  end
+
+  let(:genre)            { Faker::Book.genre }
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: You are leaving too many spaces between the let and the block

File: spec/models/book_spec.rb [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  let(:author)             { Faker::Book.author }
+  let(:image)                 { 'image' }
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: Why not use Faker gem here too?

File: spec/models/book_spec.rb [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  let(:title)              { Faker::Book.title }
+  let(:publisher)         { Faker::Book.publisher }
+  let(:year)              {'2021'}
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: Same here, why not use Faker?

File: spec/models/book_spec.rb [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+  it do
+    is_expected.to be_valid
+  end
+
+  describe '#create' do
+    context 'When the genre is nil' do
+      let(:genre) { nil }
+
+      it do
```
Comment (melissaregnier) [inline spec/models/book_spec.rb]: It is a better practice to pass a string to the it block, so it describes what it should or should not do. For example, in this case, `it 'is not valid' do`

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+    context 'When the last_name is nil' do
+      let(:last_name) { nil }
+
+      it do
+        is_expected.to be_invalid
+      end
+    end
+  end
+end
```
Comment (melissaregnier) [inline spec/models/user_spec.rb]: Missing newline at the end of the file.

**Review by melissaregnier — CHANGES_REQUESTED** [review]

File: spec/models/book_spec.rb (hunk 6 above)
Comment (NicoFunes) [inline spec/models/book_spec.rb]: you can move this to the spec/factories/books.rb with something like this.
example

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+describe User, type: :model do
+  it do
+    should validate_presence_of(:first_name)
+  end
+  it do
+    should validate_presence_of(:last_name)
+  end
+
+  subject(:user) do
```
Comment (NicoFunes) [inline spec/models/user_spec.rb:12]: same here, you can move all the user creation to the factory file

**Review by NicoFunes — CHANGES_REQUESTED** [review]

File: spec/models/book_spec.rb (hunk 9 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/user_spec.rb (hunk 10 above)
Comment (lucasenriquez) [reply spec/models/user_spec.rb]: Done

File: spec/models/book_spec.rb (hunk 6 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/user_spec.rb (hunk 11 above)
Comment (lucasenriquez) [reply spec/models/user_spec.rb:12]: Done

File: spec/models/book_spec.rb (hunk 8 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/book_spec.rb (hunk 7 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/book_spec.rb (hunk 6 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/book_spec.rb (hunk 5 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb:8]: Done

File: spec/models/book_spec.rb (hunk 4 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/models/book_spec.rb (hunk 3 above)
Comment (lucasenriquez) [reply spec/models/book_spec.rb]: Done

File: spec/factories/users.rb (hunk 2 above)
Comment (lucasenriquez) [reply spec/factories/users.rb]: Done

File: spec/factories/books.rb (hunk 1 above)
Comment (lucasenriquez) [reply spec/factories/books.rb]: Done

**Review by melissaregnier — APPROVED** [review]

**Review by NicoFunes — APPROVED** [review]

**Pull request merged**